src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java

```diff
@@ -157,6 +174,27 @@ public PaginatedSearchMetadata getPaginatedSearchMetadata(MetaData metaData, Map
         return sr;
     }
 
+    // TODO This accepts null values for backward compatibility. Once deprecated methods are removed, parameters should
+    // be annotated with @NotNull
+    private Optional<CaseDetailsEntity> find(String jurisdiction, Long id, String reference) {
+        final CaseDetailsQueryBuilder qb = queryBuilderFactory.select(em);
+
+        if (null != jurisdiction) {
+            qb.whereJurisdiction(jurisdiction);
+        }
+
+        if (null != reference) {
```

Comment:
same as above, if this is a common pattern we could have the query builder itself handle this whole if-else block to avoid duplications

Reply:
This part of the code will go away once the deprecated methods are removed. This is not the expected behaviour or a pattern that should be encouraged.